nwb_conversion_tools/datainterfaces/axonadatainterface.py

```diff
@@ -0,0 +1,136 @@
+        for bin_line in f:
+            if b'data_start' in bin_line:
+                break
+            line = bin_line.decode('cp1252').replace('\r\n', '').\
+                replace('\r', '').strip()
+            parts = line.split(' ')
+            key = parts[0]
+            if key in params:
+                header[key] = ' '.join(parts[1:])
+
+    return header
+
+
+def read_iso_datetime(set_file):
+    """
+    Creates datetime object (y, m, d, h, m, s) from .set file header
+    """
+    with open(set_file, 'r', encoding='cp1252') as f:
+        for line in f:
+            if line.startswith('trial_date'):
+                date_string = re.findall(r'\d+\s\w+\s\d{4}$', line)[0]
+            if line.startswith('trial_time'):
+                time_string = line[len('trial_time')+1::].replace('\n', '')
+
+    return datetime.datetime.strptime(date_string + ', ' + time_string,
+                                      "%d %b %Y, %H:%M:%S").isoformat()
+
+
+class AxonaRecordingExtractorInterface(BaseRecordingExtractorInterface):
+    """
+    Primary data interface class for converting a AxonaRecordingExtractor
+    """
+
+    RX = se.AxonaRecordingExtractor
+
+    @classmethod
+    def get_source_schema(cls):
+        source_schema = {
+            'required': ['filename'],
+            'properties': {
+                'filename': {
+                    'type': 'string',
+                    'format': 'file',
+                    'description': 'Path to Axona files.'
+                }
+            },
+            'type': 'object',
+            'additionalProperties': True
+        }
+        return source_schema
+
+    def __init__(self, **source_data):
+        super().__init__(**source_data)
+
+    def get_metadata(self):
+        """
+        Auto-fill metadata where possible. Must comply with metadata schema.
+        """
+
+        # Extract information for specific parameters from .set file
+        # TODO not all are currently used or part of the metadata_schema
+        params_of_interest = [
+            'experimenter',
+            'comments',
+            'duration',
+            'sw_version',
+            'tracker_version',
+            'stim_version',
+            'audio_version'
+        ]
+        set_file = self.source_data['filename']+'.set'
+        par = parse_generic_header(set_file, params_of_interest)
+
+        # Add available metadata
+        # NOTE that this interface is meant to be used within an NWBconverter,
+        # which contains a much larger metadata_schema. As such a datainterface
+        # metadata by itself does not necessarily validate with its own schema!
+        metadata = super().get_metadata()
+        metadata['NWBFile'] = dict(
+            session_start_time=read_iso_datetime(set_file),
+            session_description=par['comments'],
+            identifier=''.join(random.choices(string.ascii_uppercase +
+                                              string.digits, k=12)),
```

Comment:
Should really never have to specify this as it's handled with `uuid` in the main converter class (the thing that actually makes the NWBFiles)

Reply:
That makes sense, too. I think I just took this from some other datainterface class that I used as template: https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/datainterfaces/blackrockdatainterface.py

Reply:
Hmm, IDK what it's specified there. 

@luiztauffer do you know why you need t[hat identifier specified at the interface level](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/nwb_conversion_tools/datainterfaces/blackrockdatainterface.py#L51) instead of the automated uuid in the converter class

Reply:
@CodyCBakerPhD thanks for pointing that out, I am extracting session_start_time and comment with the data interface, the identifier apparently was inherited from an older implementation. With uuid is better indeed, I modified that

Reply:
@sbuergers if you don't need to extract that info with your datainterface, you don't need to overwrite the basic metadata automatically generated